Extension/src/LanguageServer/settings.ts

```diff
@@ -414,337 +432,253 @@ export class CppSettings extends Settings {
+        if (isArrayOfString(value)) {
+            if (!allowUndefinedEnums) {
+                if (setting.items.enum !== undefined) {
+                    if (!value.every(x => this.isValidEnum(setting.items.enum, x))) {
+                        return setting.default;
+                    }
+                }
+            }
+            return value;
+        }
+        return setting.default;
     }
 
-    public get vcFormatSpacePointerReferenceAlignment(): string {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.space.pointerReferenceAlignment")!;
+    private getAsExcludes(settingName: string): Excludes;
+    private getAsExcludes(settingName: string, allowNull: boolean): Excludes | null;
+    private getAsExcludes(settingName: string, allowNull: boolean = false): Excludes | null {
+        const value: any = super.Section.get(settingName);
+        if (allowNull && value === null) {
+            return null;
+        }
+        if (isValidMapping(value, isString, (val) => isBoolean(val) || isValidWhenObject(val))) {
+            return value as Excludes;
+        }
+        const setting = getRawSetting("C_Cpp." + settingName);
+        return setting.default as Excludes;
     }
 
-    public get vcFormatSpaceAroundTernaryOperator(): string {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.space.aroundTernaryOperator")!;
+    private getAsAssociations(settingName: string): Associations;
+    private getAsAssociations(settingName: string, allowNull: boolean): Associations | null;
+    private getAsAssociations(settingName: string, allowNull: boolean = false): Associations | null {
+        const value: any = super.Section.get<any>(settingName);
+        if (allowNull && value === null) {
+            return null;
+        }
+        if (isValidMapping(value, isString, isString)) {
+            return value as Associations;
+        }
+        const setting = getRawSetting("C_Cpp." + settingName);
+        return setting.default as Associations;
     }
 
-    public get vcFormatWrapPreserveBlocks(): string {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.wrap.preserveBlocks")!;
+    // Checks a given enum value against a list of valid enum values from package.json.
+    private isValidEnum(enumDescription: any, value: any): boolean {
+        if (isArray(enumDescription) && enumDescription.length > 0) {
+            return enumDescription.some(x => x.toLowerCase() === value.toLowerCase());
```

Comment:
@browntarik Calling ".toLowerCase()" may not be value, e.g. if a bool is used.
![image](https://github.com/user-attachments/assets/e00d14d7-ed4b-4639-b631-157397555dba)

Can you review the usage pattern of `value: any` that calls methods?